### ISSUE TITLE
Question: Source of llm.py attention mask fix?

### ISSUE THREAD

**Issue by diodiogod** [filliptm/ComfyUI_FL-CosyVoice3#7]
Hi @filliptm,

We're working on our own CosyVoice3 implementation for TTS Audio Suite and noticed that your bundled `cosyvoice/llm/llm.py` has a critical fix for attention mask handling during cached inference that isn't present in the official FunAudioLLM/CosyVoice GitHub repo.

**Your version (lines ~247-267):**
```python
# For cached inference, we need to compute the proper attention mask
# that covers the full sequence (cached positions + new positions)
if cache is not None:
    # Get the cached sequence length
    if hasattr(cache, 'get_seq_length'):
        cached_len = cache.get_seq_length()
    elif hasattr(cache, '__len__') and len(cache) > 0:
        cached_len = cache[0][0].shape[2]
    else:
        cached_len = 0
    
    total_len = cached_len + xs.shape[1]
    input_masks = torch.ones((xs.shape[0], total_len), dtype=torch.long, device=xs.device)
else:
    input_masks = masks[:, -1, :]
```

**Official repo version:**
```python
input_masks = masks[:, -1, :]
```

This fix properly handles KV cache sequence length using standard transformers patterns and resolves gibberish audio output issues we were experiencing.

**Questions:**
1. Where did this fix come from? Did you implement it yourself, or is it from a fork/PR we should reference?
2. Are there other critical fixes in your bundled CosyVoice that differ from the official repo?

We'd like to properly credit the source of this fix in our implementation. Thanks for your work on this ComfyUI wrapper!
